tonic-types/src/lib.rs

```diff
@@ -13,8 +21,444 @@
 #![doc(html_root_url = "https://docs.rs/tonic-types/0.6.0")]
 #![doc(issue_tracker_base_url = "https://github.com/hyperium/tonic/issues/")]
 
-mod pb {
-    include!(concat!(env!("OUT_DIR"), "/google.rpc.rs"));
+use prost::{DecodeError, Message};
+use prost_types::Any;
+use tonic::{codegen::Bytes, metadata::MetadataMap, Code};
+
+/// Useful protobuf types
+pub mod pb {
+    include!("generated/google.rpc.rs");
 }
 
 pub use pb::Status;
+
+mod error_details;
+mod error_details_vec;
+mod std_messages;
+
+pub use error_details::ErrorDetails;
+pub use error_details_vec::ErrorDetail;
+pub use std_messages::{BadRequest, FieldViolation};
+
+trait IntoAny {
+    fn into_any(self) -> Any;
+}
+
+trait FromAny {
+    fn from_any(any: Any) -> Result<Self, DecodeError>
+    where
+        Self: Sized;
+}
+
+fn gen_details_bytes(code: Code, message: &String, details: Vec<Any>) -> Bytes {
+    let status = pb::Status {
+        code: code as i32,
+        message: message.clone(),
+        details,
+    };
+
+    Bytes::from(status.encode_to_vec())
+}
+
+/// Used to implement associated functions and methods on `tonic::Status`, that
+/// allow the addition and extraction of standard error details.
+pub trait WithErrorDetails {
```

Comment:
What do you think about naming this `StatusExt` since its an extension trait and that way people know its methods implemented ontop of `Status`.